cfme/services/catalogs/ansible_catalog_item.py

```diff
@@ -0,0 +1,282 @@
+                "playbook": self.retirement["playbook"],
+                "machine_credential": self.retirement["machine_credential"],
+                "cloud_type": self.retirement.get("cloud_type"),
+                "hosts": self.retirement.get("hosts"),
+                "escalate_privilege": self.retirement.get("escalate_privilege"),
+                "verbosity": self.retirement.get("verbosity")
+            })
+        view.add.click()
+        view = self.create_view(AllCatalogItemView)
+        assert view.is_displayed
+        view.flash.assert_success_message("Catalog Item {} was added".format(self.name))
+
+    def update(self, updates):
+        view = navigate_to(self, "Edit")
+        changed = view.fill(updates)
+        if changed:
+            view.save.click()
+            msg = "Catalog Item {} was saved".format(updates.get("name", self.name))
+        else:
+            view.cancel.click()
+            msg = "Edit of Catalog Item {} was cancelled by the user".format(self.name)
+        view = self.create_view(DetailsAnsibleCatalogItemView, override=updates)
+        assert view.is_displayed
+        view.flash.assert_success_message(msg)
+
+    def delete(self, cancel=False):
+        view = navigate_to(self, "Details")
+        view.configuration.item_select("Remove Catalog Item", handle_alert=not cancel)
+        if cancel:
+            assert view.is_displayed
+            view.flash.assert_no_error()
+        else:
+            view = self.create_view(AllCatalogItemView)
+            assert view.is_displayed
+            view.flash.assert_success_message("The selected Catalog Item was deleted")
+
+    @property
+    def exists(self):
+        try:
+            navigate_to(self, "Details")
+        except Exception:
+            return False
+        else:
+            return True
+
+
+@navigator.register(AnsiblePlaybookCatalogItem, "All")
+class All(CFMENavigateStep):
+    VIEW = AllCatalogItemView
+    prerequisite = NavigateToAttribute("appliance.server", "LoggedIn")
+
+    def step(self):
+        self.prerequisite_view.navigation.select("Services", "Catalogs")
+        self.view.catalog_items.tree.click_path("All Catalog Items")
+
+
+@navigator.register(AnsiblePlaybookCatalogItem, "Details")
+class Details(CFMENavigateStep):
+    VIEW = DetailsAnsibleCatalogItemView
+    prerequisite = NavigateToSibling("All")
+
+    def step(self):
+        tree = self.prerequisite_view.catalog_items.tree
+        tree.click_path(
+            "All Catalog Items",
+            getattr(self.obj.catalog, "name", "Unassigned"),
+            self.obj.name
+        )
+
+
+@navigator.register(AnsiblePlaybookCatalogItem, "Add")
+class Add(CFMENavigateStep):
+    VIEW = AddAnsibleCatalogItemView
+    prerequisite = NavigateToSibling("All")
+
+    def step(self):
+        self.prerequisite_view.configuration.item_select("Add a New Catalog Item")
+        view = self.prerequisite_view.browser.create_view(SelectCatalogItemTypeView)
```

Comment:
This looks like this should be a separate step as well - One step opens the dialog with type selection and next step picks the item.